test/web-api.test.js

```diff
@@ -0,0 +1,208 @@
+  t.equal(expected.toString(), body.toString())
+})
+
+test('should response with a Response', async (t) => {
+  t.plan(3)
+
+  const fastify = Fastify()
+
+  fastify.get('/', function (request, reply) {
+    const stream = fs.createReadStream(__filename)
+    reply.send(new Response(Readable.toWeb(stream), {
+      status: 200,
+      headers: {
+        hello: 'world'
+      }
+    }))
+  })
+
+  const {
+    statusCode,
+    headers,
+    body
+  } = await fastify.inject({ method: 'GET', path: '/' })
+
+  const expected = await fs.promises.readFile(__filename)
+
+  t.equal(statusCode, 200)
+  t.equal(expected.toString(), body.toString())
+  t.equal(headers.hello, 'world')
+})
+
+test('able to use in onSend hook - ReadableStream', async (t) => {
+  t.plan(4)
+
+  const fastify = Fastify()
+
+  fastify.get('/', function (request, reply) {
+    const stream = fs.createReadStream(__filename)
+    reply.code(500).send(Readable.toWeb(stream))
+  })
+
+  fastify.addHook('onSend', (request, reply, payload, done) => {
+    t.equal(Object.prototype.toString.call(payload), '[object ReadableStream]')
+    done(null, new Response(payload, {
+      status: 200,
+      headers: {
+        hello: 'world'
+      }
+    }))
+  })
+
+  const {
+    statusCode,
+    headers,
+    body
+  } = await fastify.inject({ method: 'GET', path: '/' })
+
+  const expected = await fs.promises.readFile(__filename)
+
+  t.equal(statusCode, 200)
+  t.equal(expected.toString(), body.toString())
+  t.equal(headers.hello, 'world')
+})
+
+test('able to use in onSend hook - Response', async (t) => {
+  t.plan(4)
+
+  const fastify = Fastify()
+
+  fastify.get('/', function (request, reply) {
+    const stream = fs.createReadStream(__filename)
+    reply.send(new Response(Readable.toWeb(stream), {
+      status: 500,
+      headers: {
+        hello: 'world'
+      }
+    }))
+  })
+
+  fastify.addHook('onSend', (request, reply, payload, done) => {
+    t.equal(Object.prototype.toString.call(payload), '[object Response]')
+    done(null, new Response(payload.body, {
+      status: 200,
+      headers: payload.headers
+    }))
+  })
+
+  const {
+    statusCode,
+    headers,
+    body
+  } = await fastify.inject({ method: 'GET', path: '/' })
+
+  const expected = await fs.promises.readFile(__filename)
+
+  t.equal(statusCode, 200)
+  t.equal(expected.toString(), body.toString())
+  t.equal(headers.hello, 'world')
+})
+
+test('Error when Response.bodyUsed', async (t) => {
+  t.plan(4)
+
+  const expected = await fs.promises.readFile(__filename)
+
+  const fastify = Fastify()
+
+  fastify.get('/', async function (request, reply) {
+    const stream = fs.createReadStream(__filename)
+    const response = new Response(Readable.toWeb(stream), {
+      status: 200,
+      headers: {
+        hello: 'world'
+      }
+    })
+    const file = await response.text()
+    t.equal(expected.toString(), file)
+    t.equal(response.bodyUsed, true)
+    return reply.send(response)
```

Comment:
just to handle this use case too

```suggestion
    return response
```

Reply:
Doesn't the test below already handled the `Promise` return use-case?